microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -79,6 +90,60 @@ def run_tests(
 
         assert_that(failures, f"Unexpected failures: {failures}").is_empty()
 
+    def run_metrics_tests(
+        self,
+        test_result: TestResult,
+        environment: Environment,
+        hypervisor: str,
+        skip: Optional[List[str]] = None,
+    ) -> None:
+        temp = self.repo_root.joinpath("perf_mtr_report")
+        self.per_mtr_report_file = temp.joinpath("testcase_result.json")
+
+        perf_metrics_tests = self._list_perf_metrics_tests(hypervisor=hypervisor)
+        testcase_result = {}
+        for testcase in perf_metrics_tests:
+            self._log.info(f"Running testcase : {testcase}")
+            testcase_result[testcase] = {}
+            testcase_result[testcase]["status"] = "Started"
+
+            try:
+                result = self.run(
+                    f"tests --hypervisor {hypervisor} --metrics -- -- \
+                        --test-filter {testcase}",
+                    timeout=self.TIME_OUT,
+                    force_run=True,
+                    cwd=self.repo_root,
+                    no_info_log=False,  # print out result of each test
+                    shell=True,
+                )
+                output = result.stdout.replace("\r\n", "\n")
+                output = output.replace("\t", "")
+                if result.exit_code != 0:
+                    excep = Exception(f"Test Case Failed : {testcase}", output)
+                    raise excep
+                testcase_result[testcase]["status"] = "Succeeded"
+                metrics = self._process_perf_metric_test_result(result.stdout)
+                testcase_result[testcase]["metrics"] = metrics
+                testcase_result[testcase]["err"] = ""
+            except Exception as e:
+                self._log.error(f"Testcase failed , tescase name : {testcase}")
+                testcase_result[testcase]["status"] = "Failed"
+                testcase_result[testcase]["err"] = str(e.args[0])
+                testcase_result[testcase]["trace"] = str(e.args[1])
+                testcase_result[testcase]["metrics"] = ""
+        report = self._create_perf_metric_report(testcase_result)
```

Comment:
Why not directly create the `CHPerfMetricTestResult` object instead of first creating an intermediate dictionary and then creating a `CHPerfMetricTestResult` out of that?

Reply:
sure , will work on this

Reply:
Implementes, please check